src/pivot-table/data/helpers/create-cell.ts

```diff
@@ -10,26 +15,49 @@ const createCell = (
   pageY: number,
   isSnapshot: boolean,
   dimensionInfo: VisibleDimensionInfo,
-): Cell => ({
-  ref: node,
-  x,
-  y,
-  // pageX might change to reflect x in current x axis page
-  // when we implement horizontal pagination feature (exactly like the relation btw y and pageY)
-  pageX: x,
-  pageY,
-  parent,
-  root,
-  leafCount: isSnapshot ? 0 : node.qUp + node.qDown,
-  distanceToNextCell: 0,
-  isLockedByDimension: typeof dimensionInfo === "object" && dimensionInfo.qLocked,
-  incrementLeafCount() {
-    this.leafCount += 1;
-    if (parent) {
-      parent.incrementLeafCount();
-    }
-  },
-  isTotalCell: node.qType === NxDimCellType.NX_DIM_CELL_TOTAL || !!parent?.isTotalCell,
-});
+): Cell => {
+  const cell = {
+    ref: node,
+    x,
+    y,
+    // pageX might change to reflect x in current x axis page
+    // when we implement horizontal pagination feature (exactly like the relation btw y and pageY)
+    pageX: x,
+    pageY,
+    parent,
+    root,
+    children: [],
+    leafCount: isSnapshot ? 0 : node.qUp + node.qDown,
+    distanceToNextCell: 0,
+    isLockedByDimension: !!(typeof dimensionInfo === "object" && dimensionInfo.qLocked),
+    incrementLeafCount() {
+      this.leafCount += 1;
+      if (parent) {
+        parent.incrementLeafCount();
+      }
+    },
+    // Having "parent.isTotal" means that it's enough that any ancestors is a total cell,
+    // which is needed for the Total cell highlight use case.
+    isTotal: node.qType === NxDimCellType.NX_DIM_CELL_TOTAL || !!parent?.isTotal,
+    isEmpty: node.qType === NxDimCellType.NX_DIM_CELL_EMPTY,
+    isNull: node.qType === NxDimCellType.NX_DIM_CELL_NULL,
+    isPseudoDimension: node.qType === NxDimCellType.NX_DIM_CELL_PSEUDO,
+    // A getter because child nodes are added as cells are being created. It has to be resolved when it's called.
+    get isLastChild(): boolean {
+      // Root is considedered last child for the total cell divider use case
+      if (root === null) {
+        return true;
+      }
+
+      // Having "parent.isLastChild" means that all ancestors also have to be the last child,
+      // which is needed for the Total cell divider use case.
+      return parent?.children[parent.children.length - 1] === cell && parent.isLastChild;
```

Comment:
is `cell` refers to the current cell object that is being created? then how this condition could be possible? 
```tsx
parent?.children[parent.children.length - 1] === cell
```

even if both objects will having the same key/values, it will be false because cell is being generated every time that js interpreter reaches this line. (Or am i missing something?)

<img width="184" alt="Screenshot 2023-08-29 at 10 32 07" src="https://github.com/qlik-oss/sn-pivot-table/assets/29652890/06a82483-f46d-4724-a24b-fcce8bf97f7e">

Reply:
I think the commet for `isLastChild` tries to describe why: "// A getter because child nodes are added as cells are being created. It has to be resolved when it's called."

I simplified version of it should look something like this:
```js
const parent = { isLastChild: true, children: [] );
const firstCell = { isLastChild: () => parent.children.at(-1) === firstCell && parent.isLastChild };
const secondCell = { isLastChild: () => parent.children.at(-1) === secondCell && parent.isLastChild };

parent.children.push(firstCell);
firstCell.isLastChild(); // => TRUE, but this is not when it's executed.

parent.children.push(secondCell);
// All cells have been added. The cells are now rendered and "isLastChild" is called.
firstCell.isLastChild(); // => FALSE
secondCell.isLastChild(); // => TRUE
```

Does that clear it up a little bit?

Reply:
Or maybe you are thinking about closure and scope?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures#closure

Reply:
Yes, it is clean now. 

Basically what i understood is: the cell within each of `isLastChild` function (which exists within the same cell) creates a closure that holds a reference to the same cell. 

and when you execute that function, it checks the same reference that it's closure holds.

What i was confused was with the checking part with cell (`=== cell` part), then I questioned my self why not using `this` here which is more not confusing (at least for me). 

But now i figured out why since you are using arrow function, `this` will point to window object (I believe) and not the correct reference to `cell` object. But it works if you convert `isLastChild` to typical `function() {...}` call

but i'll leave that judgment up to you to see which one is better to go with in this case, and of course its LGTM now 🚢